src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDisplayClassOptimisationTests.cs

```diff
@@ -0,0 +1,5414 @@
+  IL_0012:  dup
+  IL_0013:  ldstr      ""one""
+  IL_0018:  callvirt   ""void System.Collections.Generic.List<string>.Add(string)""
+  IL_001d:  stfld      ""System.Collections.Generic.List<string> Program.<>c__DisplayClass0_0.strings""
+  IL_0022:  ldloc.0
+  IL_0023:  newobj     ""Program.Disposable..ctor()""
+  IL_0028:  stfld      ""Program.Disposable Program.<>c__DisplayClass0_0.disposable""
+  .try
+  {
+    IL_002d:  ldloc.1
+    IL_002e:  ldloc.0
+    IL_002f:  ldc.i4.0
+    IL_0030:  stfld      ""int Program.<>c__DisplayClass0_0.i""
+    IL_0035:  ldloc.0
+    IL_0036:  ldftn      ""void Program.<>c__DisplayClass0_0.<Main>b__0()""
+    IL_003c:  newobj     ""System.Action..ctor(object, System.IntPtr)""
+    IL_0041:  callvirt   ""void System.Collections.Generic.List<System.Action>.Add(System.Action)""
+    IL_0046:  leave.s    IL_005c
+  }
+  finally
+  {
+    IL_0048:  ldloc.0
+    IL_0049:  ldfld      ""Program.Disposable Program.<>c__DisplayClass0_0.disposable""
+    IL_004e:  brfalse.s  IL_005b
+    IL_0050:  ldloc.0
+    IL_0051:  ldfld      ""Program.Disposable Program.<>c__DisplayClass0_0.disposable""
+    IL_0056:  callvirt   ""void System.IDisposable.Dispose()""
+    IL_005b:  endfinally
+  }
+  IL_005c:  ldloc.1
+  IL_005d:  ldc.i4.0
+  IL_005e:  callvirt   ""System.Action System.Collections.Generic.List<System.Action>.this[int].get""
+  IL_0063:  callvirt   ""void System.Action.Invoke()""
+  IL_0068:  ret
+}");
+
+            compilation.VerifyTypeIL("<>c__DisplayClass0_0", @"
+.class nested private auto ansi sealed beforefieldinit '<>c__DisplayClass0_0'
+	extends [mscorlib]System.Object
+{
+	.custom instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = (
+		01 00 00 00
+	)
+	// Fields
+	.field public class [mscorlib]System.Collections.Generic.List`1<string> strings
+	.field public class Program/Disposable disposable
+	.field public int32 i
+	// Methods
+	.method public hidebysig specialname rtspecialname 
+		instance void .ctor () cil managed 
+	{
+		// Method begins at RVA 0x20da
+		// Code size 7 (0x7)
+		.maxstack 8
+		IL_0000: ldarg.0
+		IL_0001: call instance void [mscorlib]System.Object::.ctor()
+		IL_0006: ret
+	} // end of method '<>c__DisplayClass0_0'::.ctor
+	.method assembly hidebysig 
+		instance void '<Main>b__0' () cil managed 
+	{
+		// Method begins at RVA 0x20e2
+		// Code size 39 (0x27)
+		.maxstack 8
+		IL_0000: ldarg.0
+		IL_0001: ldfld class Program/Disposable Program/'<>c__DisplayClass0_0'::disposable
+		IL_0006: callvirt instance string [mscorlib]System.Object::ToString()
+		IL_000b: call void [mscorlib]System.Console::WriteLine(string)
+		IL_0010: ldarg.0
+		IL_0011: ldfld class [mscorlib]System.Collections.Generic.List`1<string> Program/'<>c__DisplayClass0_0'::strings
+		IL_0016: ldarg.0
+		IL_0017: ldfld int32 Program/'<>c__DisplayClass0_0'::i
+		IL_001c: callvirt instance !0 class [mscorlib]System.Collections.Generic.List`1<string>::get_Item(int32)
+		IL_002
```

Comment:
So it looks like `b` isn't in this display class -- can you dump `Program` instead? That should show everything